libs/ccf-shared/src/lib/services/data-source/api-endpoint.service.ts

```diff
@@ -101,11 +104,39 @@ function spatialSceneNodeReviver(nodes: RawSpatialSceneNode[]): SpatialSceneNode
   providedIn: 'root',
 })
 export class ApiEndpointDataSourceService implements DataSource {
+  private readonly http = inject(HttpClient);
+
   constructor(
     private readonly api: DefaultService,
     private readonly globalConfig: GlobalConfigState<ApiEndpointDataSourceOptions>,
   ) {
     globalConfig.getOption('hubmapToken').subscribe(buster$);
+    combineLatest([
+      globalConfig.getOption('dataSources'),
+      globalConfig.getOption('filter'),
+      globalConfig.getOption('remoteApiEndpoint'),
+    ])
+      .pipe(
+        tap(([sources, filter, endpoint]) => {
+          if ((sources && sources.length > 0) || filter) {
+            const sessionTokenRequest = {
+              dataSources: sources,
+              filter: filter,
+            };
+            console.log(sessionTokenRequest);
+
+            // Get session token
+            this.http.post(endpoint + '/session-token', sessionTokenRequest).subscribe((resp: DefaultParams) => {
```

Comment:
You'll want to catch errors here as the endpoint might not have /session-token and then you'll just need to set the filter to ''. May need to do that for other cases here as well.